### PR TITLE
Issue 11

### PR DESCRIPTION
**Pull request by mgazz**
With these changes the patch method takes as input only the changes that we want to apply to the target object. I also updated the test to fit the new mechanism.

See #11  for a more detailed description of the problem.

### CODE REVIEW

**Comment by mgazz** [comment]
@christian-pinto thank you for the review. I rebased the branch and added verified commits. I'll be available in case on any question.

**Comment by christian-pinto** [comment]
@mjaguil could you please have a look at this PR?

File: sunfish/lib/core.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def replace_object(self, payload):
         # Call function from backend
         return self.storage_backend.replace(payload)
     
-    def patch_object(self, payload):
+    def patch_object(self, path,payload):
```
Comment (dledford) [inline sunfish/lib/core.py:118]: Whitespace inconsistency is annoying.  Doubly so when you add an inconsistency in the middle of a single statement :-/

File: sunfish/lib/core.py [hunk 2]
```diff
@@ -135,7 +135,7 @@ def patch_object(self, payload):
             return resp
                     
         # call function from backend
-        return self.storage_backend.patch(payload)
+        return self.storage_backend.patch(path,payload)
```
Comment (dledford) [inline sunfish/lib/core.py:138]: More here

File: sunfish/lib/exceptions.py [hunk 3]
```diff
@@ -17,6 +17,23 @@ def __init__(self, resource_id):
...
+        Exception raised when the resource is not found.
+
+        Attributes:
+        attribute_name -- name of the attribute not found
+        message -- explanation of the error
++    """
+
+    def __init__(self, attribute_name):
+        self.attribute_name=attribute_name
+        message = "Attribute " + attribute_name+ " not found."
```
Comment (dledford) [inline sunfish/lib/exceptions.py:32]: More here.

File: sunfish/storage/backend_FS.py [hunk 4]
```diff
@@ -157,9 +157,11 @@ def replace(self, payload:json):
         except ResourceNotFound as e:
             raise ResourceNotFound(e.resource_id)
     
-    def patch(self, payload:json):
+    def patch(self, path: str, payload:json):
```
Comment (dledford) [inline sunfish/storage/backend_FS.py:160]: More here

**Review by dledford — CHANGES_REQUESTED** [review]
Comments on the respective code lines

**Comment by mgazz** [comment]
@dledford thank you for the feedback. I will update as spacing to make it more consistent.

File: sunfish/lib/core.py (hunk 1 above)
Comment (mgazz) [reply sunfish/lib/core.py]: Fixed in 4e3226cd76e3e06811f6c4e58bb13ab288c8e198

File: sunfish/lib/core.py (hunk 2 above)
Comment (mgazz) [reply sunfish/lib/core.py]: Fixed in 4e3226cd76e3e06811f6c4e58bb13ab288c8e198

File: sunfish/lib/exceptions.py (hunk 3 above)
Comment (mgazz) [reply sunfish/lib/exceptions.py]: Fixed in d102571eb4fcbf0640d50cfd77489af740e0275a

File: sunfish/storage/backend_FS.py (hunk 4 above)
Comment (mgazz) [reply sunfish/storage/backend_FS.py]: Fixed in d102571eb4fcbf0640d50cfd77489af740e0275a

**Review by dledford — APPROVED** [review]
Thanks for updating the style issues

**Review by christian-pinto — COMMENTED** [review]
Looks good to me

**Review by christian-pinto — APPROVED** [review]

**Pull request merged**